openssh/config.sls

```diff
@@ -3,17 +3,35 @@
 include:
   - openssh
 
-{% if salt['pillar.get']('sshd_config', False) %}
-sshd_config:
+{%- macro sshd_config_file(name) %}
   file.managed:
-    - name: {{ openssh.sshd_config }}
+    - name: {{ name }}
     - source: {{ openssh.sshd_config_src }}
     - template: jinja
     - user: {{ openssh.sshd_config_user }}
     - group: {{ openssh.sshd_config_group }}
     - mode: {{ openssh.sshd_config_mode }}
     - watch_in:
       - service: {{ openssh.service }}
+{%- endmacro %}
+
+{% if salt['pillar.get']('sshd_config', False) %}
+{% set test_config = openssh.sshd_config+'.saltstack-test' %}
```

Comment:
I'd move the test file to a temp/cache dir, just to keep the `/etc/ssh/` dir clean. Perhaps you can set a variable (say, `sshd_config_test_file`) directly in the [defaults.yaml](https://github.com/saltstack-formulas/openssh-formula/blob/master/openssh/defaults.yaml#L3) pointing initially to some place like `/var/tmp/sshd.saltstack-test` or `/var/cache/sshd.saltstack-test` ?

What do you think?